### DISCUSSION TITLE
Debugging bad composer installation without internet connection

### DISCUSSION THREAD

**Discussion by hdevacc** [composer/composer#11382] [Q&A / Support]
I have a legacy app that is installed on a computer without internet access.
Now because of company policies it would be very hard to try to do any work around such as using the `composer.json` to do a fresh install and then put it back.

I was not the one who installed the app (Laravel 6 app) and I still don't know what is the cause of the problem, but I currently have this situation:

The app is working fine with the current `vendor` folder. However when I want to run any composer command such as `composer dump-autoload` it gives me an error message:

> Class 'adldap laravel adldapserviceprovider' not found

Then the app completely breaks, when I try to enter the URL I get `unable to reach` error in the browser.

So what I did was to add a line to the `require` part of `composer.json`:

    "adldap2/adldap2-laravel": "3.0.*"

Then when I ran `composer dump-autoload`, it ran successfully and the app was working again and I no longer got the error above with the composer commands.

But this messed other parts of the app, such as not finding a class that was found with the original `vendor` folder, even though the files themselves exist.

When I copy again the back `vendor` folder the app is fully working again, but then I can't run `composer dump-autoload` again.

How can I debug the cause and try to make it work with the `composer dump-autoload` command, and also keep the other parts working?

I assume that all of these errors come from the `composer` folder inside `vendor` because it can't change other folders there since there is no internet connection.

So maybe some files there were manually edited.
But how can I try to find out what are the errors?
Why does the app work fine until I run `composer dump-autoload`, and why does it break other parts?

I've been trying to find the cause but I'm not sure which parts in `vendor` to look for. I know it's fixable because with the original `vendor` the app is working fine

I want to be able to run composer dump-autoload without breaking the app

Or I should just leave the original vendor folder as it is without touching it anymore?

I just feel that it might be some problems in one of the files in composer folder in vendor and that it's fixable

**Comment by Seldaek** [comment]
Could you run a `composer install --dry-run` to see if that would change anything? I'm thinking maybe some packages are marked installed but not required or some sort of inconsistent state leading to your problems, maybe this would give us a hint.. 

Is there any chance to copy the files somewhere else tho to try and figure it out online, do a dependency update perhaps, then copy the files back?
